slides/pharma-rug-mmrm-mar2023.qmd

```diff
@@ -0,0 +1,339 @@
+-   Developing high-quality software with good coding practices, reproducible outputs, and self-sufficient documentation is critical to inform clinical and regulatory decisions
+
+## How to deal with these issues?
+
+-   To deal with the issues of statistical quality assurances for R packages and creating high quality statistical software a group came together to create through the "Software Engineering Working Group" (SWE WG) to look R packages from a statistical point of view
+
+## Software Engineering working group (SWE WG)
+
+-   An official working group of the ASA Biopharmaceutical Section
+-   Formed in August 2022
+-   Cross-industry collaboration with more than 30 members from over 20 organizations
+-   Home page at [rconsortium.github.io/asa-biop-swe-wg](https://rconsortium.github.io/asa-biop-swe-wg)
+
+## Why do we need a package for MMRM?
+::: columns
+::: {.column width="50%"}
+-   Mixed Models for Repeated Measures (MMRM) is a popular choice for analyzing longitudinal continuous outcomes in randomized clinical trials
+-   No great R Package - initially though that the MMRM problem was solved by using a combination of `lme4` and `lmerTest`
+:::
+
+
+::: {.column width="50%"}
+-   Learned that this approach failed on large data sets (slow, did not converge)
+-   `nlme` does not give Satterthwaite adjusted degrees of freedom, has convergence issues, and with `emmeans` it is only approximate
+
+:::
+:::
+
+## Before creating a new package
+
+- First try to improve existing package
+  - Here we tried to extend `glmmTMB` to calculate Satterthwaite adjusted degrees of freedom
+  - But it did not work
+    - `glmmTMB` is always using a random effects representation, we cannot have a real unstructured model (uses $\sigma = \varepsilon > 0$ trick)
+- Think about long term maintenance and responsibility
+
+## Idea with some Details
+
+-   We only want to fit a fixed effects model with a structured covariance matrix for each subject
+-   The idea is then to use the Template Model Builder (`TMB`) directly - as it is also underlying `glmmTMB` - but code the exact model we want
+-   We do this by implementing the log-likelihood in `C++` using the `TMB` provided libraries
+
+## Advantages of `TMB`
+
+-   Fast `C++` framework for defining objective functions (`Rcpp` would have been alternative interface)
+-   Automatic differentiation of the log-likelihood as a function of the variance parameters
+-   Syntactic sugars to allow simple matrix calculations or operations like R
+-   We get the gradient and Hessian exactly and without additional coding
+-   This can be used from the R side with the `TMB` interface and plugged into optimizers
+
+## Why it's not just another package
+
+-   Ongoing maintenance and support from the pharmaceutical industry
+    -   5 companies being involved in the development, on track to become standard package
```

Comment:
check number here, maybe it is already 6 now or so :-)

Reply:
in the description page, only BI, Gilead, Roche, Merck and AZ